src/registrar/admin.py

```diff
@@ -224,6 +226,31 @@ class ContactAdmin(ListHeaderAdmin):
     search_help_text = "Search by firstname, lastname or email."
 
 
+class DomainApplicationAdminForm(forms.ModelForm):
+    """Custom form to limit transitions to available transitions"""
+
+    class Meta:
+        model = models.DomainApplication
+        fields = "__all__"
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        instance = kwargs.get("instance")
+        if instance and instance.pk:
+            current_state = instance.status
+            transitions = get_available_FIELD_transitions(
+                instance, models.DomainApplication._meta.get_field("status")
+            )
+            # first option in status transitions is current state
+            available_transitions = [(current_state, current_state)]
```

Comment:
(optional) From a code readability standpoint, I'd move this to line 242 (above transitions)

Reply:
done; i moved the code for readability